### PR TITLE
Kovan+Rinkeby deploy scripts

### PR DESCRIPTION
**Pull request by krzkaczor**
```
Wormhole:  {
  "join": "0x5B321180cC155a6fd38bc14a64205d1344317975",
  "oracleAuth": "0xcEBe310e86d44a55EC6Be05e0c233B033979BC67",
  "router": "0x7e178860F560c8eb8a493113bDeB23A5db8B945F",
  "constantFee": "0x2aE3853F2B7a410e2789D7Dd38D42FC63eB982e5",
  "relay": "0x76bf2A5293A03aE921a2c6eE27B28759aaaAad64"
}
Optimism wormhole bridge:  {
  "l2WormholeBridge": "0x45440Ae4988965A4cD94651E715fC9A04e62Fb41",
  "l1WormholeBridge": "0x646231dFDAF583E2eB77FEdc47433C30519FF448"
}
Spells: {
  "l1": "0x66b3d63621fdd5967603a824114da95cc3a35107",
  "l2": "0x992C01191D62C0C333ef23935978749B50eDbC82"
}
```

### CODE REVIEW

File: deployment/deploy-kovan.ts [hunk 1]
```diff
@@ -1,8 +1,111 @@
...
+import { deployWormhole } from '../test/wormhole'
+import { performSanityChecks } from '../test/wormhole/checks'
+
+const bytes32 = ethers.utils.formatBytes32String
+
+async function main() {
+  const fee = 0 // 0 fees
+  const feeTTL = 60 * 60 * 24 * 8 // flush should happen more or less, 1 day after initWormhole, and should take 7 days to finalize
+  const ilk: string = bytes32('WH-KOVAN-TEST-1')
+  const masterDomain = bytes32('KOVAN-MASTER-1')
```
Comment (olivdb) [inline deployment/deploy-kovan.ts:20]: Should we name this `KOVAN-MASTER` as there won't be more than one master domain per network (unlike for slave domains)?

Comment (olivdb) [reply deployment/deploy-kovan.ts:20]: Also, is the MASTER/SLAVE necessary in domain ids? We could simply use `KOVAN` and `KOVAN-OPTIMISM-1`.

File: test/helpers/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export async function deployUsingFactoryAndVerify<T extends ContractFactory>(
+  signer: Signer,
+  factory: T,
+  args: Parameters<T['deploy']>,
+): Promise<ReturnType<T['deploy']>> {
+  const contractDeployed = await deployUsingFactory(signer, factory, args)
+
+  console.log(
+    `npx hardhat verify ${contractDeployed.address} ${args
```
Comment (olivdb) [inline test/helpers/deploy.ts:45]: Could we also execute this command via a child process?

File: test/helpers/txs.ts [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+  ) {
+    return defaultWhenReorgsPossible
+  } else if (chainId === 69 || chainId === 10) {
+    // optimism needs *some* delay
+    return 1
+  } else if (chainId === 421611) {
+    // arbitrum needs *some* delay
+    return 1
+  } else {
+    return defaultForInstantFinality
```
Comment (olivdb) [inline test/helpers/txs.ts:30]: Shouldn't the default be 1? Waiting for zero confirmation is the same as not calling `wait()` I think.

File: deployment/deploy-kovan.ts (hunk 1 above)
Comment (krzkaczor) [reply deployment/deploy-kovan.ts:20]: > Should we name this KOVAN-MASTER as there won't be more than one master domain per network (unlike for slave domains)?

I dont' consider this deployment (even master) final one. I think we will add some small tweaks...

> Also, is the MASTER/SLAVE necessary in domain ids? We could simply use KOVAN and KOVAN-OPTIMISM-1.

I wanted to be explicit 🤷  I don't have a strong opinion on this.

File: test/helpers/txs.ts (hunk 3 above)
Comment (krzkaczor) [reply test/helpers/txs.ts:30]: hmm maybe? I'll try changing this

File: test/helpers/deploy.ts (hunk 2 above)
Comment (krzkaczor) [reply test/helpers/deploy.ts:45]: It's much harder than that, unfortunately:
1. You need to specify network argument that matches the one from hardhat config
2. Artifacts are not in this hh project so this even doesn't work and I ended up validating contracts manually :/
3. This is mostly useful to later manually tweak CLI cmd or get ctor args.

File: test/wormhole/setup.ts [hunk 4]
```diff
@@ -115,7 +115,7 @@ export async function setupTest({
   const ilk: string = bytes32('WH_' + Buffer.from(randomBytes(14)).toString('hex'))
 
   const {
-    l1Sdk,
```
Comment (olivdb) [inline test/wormhole/setup.ts:118]: We can use `makerSdk` instead of `l1Sdk` in this file

File: test/helpers/txs.ts (hunk 3 above)
Comment (krzkaczor) [reply test/helpers/txs.ts:24]: Good catch. I thought `1` will be stuck on local chain....

**Review by olivdb — APPROVED** [review]

**Review by olivdb — APPROVED** [review]

**Pull request merged**